### PR TITLE
Fix extraneous skipped seq no

### PR DESCRIPTION
**Pull request by dcristoloveanu**
- Improve hash table int tests to check that all sequence numbers are actually indicated
- Improve sorted list int tests to check that all sequence numbers are actually indicated
- Fix extraneous skipped seq no in insert/delete for sorted list

### CODE REVIEW

File: tests/clds_hash_table_int/clds_hash_table_int.c [hunk 1]
```diff
@@ -1637,24 +1658,51 @@ static int chaos_thread(void* arg)
...
     CHAOS_TEST_CONTEXT* chaos_test_context = (CHAOS_TEST_CONTEXT*)malloc(sizeof(CHAOS_TEST_CONTEXT) + (sizeof(CHAOS_TEST_ITEM_DATA) * CHAOS_ITEM_COUNT));
     ASSERT_IS_NOT_NULL(chaos_test_context);
 
+    for (i = 0; i < TEST_SEQ_NO_QUEUE_SIZE; i++)
+    {
+        (void)InterlockedExchange(&chaos_test_context->seq_numbers[i], SEQ_NO_NOT_USED);
+    }
+
     (void)InterlockedExchange64(&chaos_test_context->seq_no_count, 0);
+    (void)InterlockedExchange64(&chaos_test_context->next_seq_no_to_ack, 1);
```
Comment (mattdurak) [inline tests/clds_hash_table_int/clds_hash_table_int.c:1698]: >next_seq_no_to_ack [](start = 53, length = 18)

This isn't volatile and isn't using interlocked in the other thread. Should it just be a local?

Comment (mattdurak) [reply tests/clds_hash_table_int/clds_hash_table_int.c:1698]: Was also wondering if something like this should check that we don't go all the way around the queue and overwrite things (probably unlikely with the size of the queue though...)

---
In reply to: [468054641](https://github.com/Azure/clds/pull/7#discussion_r468054641) [](ancestors = 468054641)

File: tests/clds_sorted_list_int/clds_sorted_list_int.c [hunk 2]
```diff
@@ -1497,22 +1545,51 @@ static int chaos_thread(void* arg)
...
     CHAOS_TEST_CONTEXT* chaos_test_context = (CHAOS_TEST_CONTEXT*)malloc(sizeof(CHAOS_TEST_CONTEXT) + (sizeof(CHAOS_TEST_ITEM_DATA) * CHAOS_ITEM_COUNT));
     ASSERT_IS_NOT_NULL(chaos_test_context);
 
+    for (i = 0; i < TEST_SEQ_NO_QUEUE_SIZE; i++)
+    {
+        (void)InterlockedExchange(&chaos_test_context->seq_numbers[i], SEQ_NO_NOT_USED);
+    }
+
+    (void)InterlockedExchange64(&chaos_test_context->seq_no_count, 0);
+    (void)InterlockedExchange64(&chaos_test_context->next_seq_no_to_ack, 1);
```
Comment (mattdurak) [inline tests/clds_sorted_list_int/clds_sorted_list_int.c:1585]: >InterlockedExchange64 [](start = 10, length = 21)

same comment about not using interlocked/volatile

File: tests/clds_hash_table_int/clds_hash_table_int.c (hunk 1 above)
Comment (dcristoloveanu) [reply tests/clds_hash_table_int/clds_hash_table_int.c:1698]: Fixed.

---
In reply to: [468057928](https://github.com/Azure/clds/pull/7#discussion_r468057928) [](ancestors = 468057928,468054641)

File: tests/clds_sorted_list_int/clds_sorted_list_int.c (hunk 2 above)
Comment (dcristoloveanu) [reply tests/clds_sorted_list_int/clds_sorted_list_int.c:1585]: Fixed.

---
In reply to: [468058351](https://github.com/Azure/clds/pull/7#discussion_r468058351) [](ancestors = 468058351)

File: tests/clds_hash_table_int/clds_hash_table_int.c (hunk 1 above)
Comment (dcristoloveanu) [reply tests/clds_hash_table_int/clds_hash_table_int.c:1698]: For the second part as discussed in person, the count of indicated sequence numbers is also checked. Thus it is not really possible to skip a sequence number, wrap around and thus mask a bug.

---
In reply to: [468101203](https://github.com/Azure/clds/pull/7#discussion_r468101203) [](ancestors = 468101203,468057928,468054641)

**Pull request merged**